ibm_vpd_app.cpp

```diff
@@ -127,11 +127,22 @@ static auto isFruInVpdCache(const std::string& objectPath)
  */
 static auto needsRecollection(const nlohmann::json& js, const string& filePath)
 {
-    if (js["frus"][filePath].at(0).value("pluggableAtStandby", false) ||
-        js["frus"][filePath].at(0).value("concurrentlyMaintainable", false))
+    if (js["frus"][filePath].at(0).value("pluggableAtStandby", false))
     {
         return true;
     }
+
+    const std::vector<nlohmann::json>& groupEEPROM =
+        js["frus"][filePath].get_ref<const nlohmann::json::array_t&>();
+
+    for (const auto& item : groupEEPROM)
+    {
+        if (item.find("concurrentlyMaintainable") != item.end())
```

Comment:
can you also check the value of `concurrentlyMaintainable` key?
We can't assume that this key will be hardcoded in json only with a value true.

Reply:
Fixed.